mlflow/sklearn/__init__.py

```diff
@@ -513,19 +513,13 @@ def predict(
         """
         return self.sklearn_model.predict(data)
 
-    def predict_proba(
-        self, data, params: Optional[Dict[str, Any]] = None  # pylint: disable=unused-argument
-    ):
-        """
-        :param data: Model input data.
-        :param params: Additional parameters to pass to the model for inference.
+    def predict_proba(self, *args, **kwargs):
+        if hasattr(self.sklearn_model, "predict_proba"):
```

Comment:
Do we need this condition?

Reply:
This means this function returns None, which can lead to a hard-to-debug error. Just throwing `sklearn_model has no attribute predict_proba` might easier to debug.

Reply:
Yes, for this test: https://github.com/mlflow/mlflow/actions/runs/5527963781/jobs/10084241551#step:8:3032
on commit https://github.com/mlflow/mlflow/pull/9050/commits/e06bf73859a6682a4d39db41d452b08d207fa9ee
```
FAILED tests/evaluate/test_default_evaluator.py::test_svm_classifier_evaluation_disable_logging_metrics_and_artifacts - AttributeError: 'LinearSVC' object has no attribute 'predict_proba'
```

Reply:
Interesting, how did this test pass before you made the inference params change?

Reply:
Because `self.predict_fn, self.predict_proba_fn = _extract_predict_fn(model, self.raw_model)`. If the raw_model is the underlying sklearn_model, then from here it is the sklearn model's predict_proba function, instead of the wrapper's predict_proba. But if raw_model is a _SklearnModelWrapper, then `self.predict_prob_fn` always exist, we might need to update the logic here again since `predict_proba_fn = getattr(raw_model, "predict_proba", None)` is not None even the function itself returns None 🤔

Reply:
Or it's fine, because 
```
if self.predict_proba_fn is not None:
    self.y_probs = self.predict_proba_fn(self.X.copy_to_avoid_mutation())
else:
    self.y_probs = None
```
in the first situation it returns None as well